src/app/shared/data/allocations.service.ts

```diff
@@ -33,16 +33,37 @@ export class AllocationsService {
     return this.allocationsSubject$.getValue();
   }
 
-  addStudentToProject(studentId: string, projectId: string): void {
+  get allocations$(): Observable<Allocation[]> {
+    return this.allocationsSubject$.asObservable();
+  }
+
+  moveStudentToProject(studentId: string, projectId: string): void {
     this.removeStudentFromProjects(studentId);
 
     const allocations = this.getAllocations();
     const allocation = allocations.find(a => a.projectId === projectId);
 
     if (allocation) {
       allocation.students.push(studentId);
+    } else {
+      allocations.push({ projectId, students: [studentId] });
+    }
+    this.setAllocations(allocations);
+  }
+
+  moveStudentToProjectAtInset(studentId: string, projectId: string, siblingId?: string): void {
+    this.removeStudentFromProjects(studentId);
```

Comment:
The following code is used in `moveStudentToProjectAtInset` and ` moveStudentToProject`. 
```
  this.removeStudentFromProjects(studentId);

    const allocations = this.getAllocations();
    const allocation = allocations.find(a => a.projectId === projectId);
```
Create a new function that you can use both times.